src/draw/DefaultDrawer.js

```diff
@@ -695,87 +783,109 @@ class DefaultDrawer {
   setupTiles(lines) {
     let previousTallestItem = { x1: 0, y1: 0 };
 
-    lines.forEach((line, lineIndex) => {
-      let prevItem = {
-        x1: 0,
-        y0: lineIndex * this.minHeight + this.padding,
-      };
-
-      line.items.forEach((item) => {
-        if (item.data.drawOption) {
-          item.x0 = item.data.drawOption.x;
-          item.y0 = item.data.drawOption.y;
-        } else {
-          item.x0 = prevItem.x1 + this.padding;
-          item.y0 = previousTallestItem.y1 + this.padding;
-          prevItem = item;
-        }
-
-        if (!item.data.drawOption || item.data.drawOption.needsResizing) {
-          /*
-           item.depth and item.height are set by d3 and represent the position of the node in
-           the hierarchy:
-           - height: how many layers exist below this node;
-           - depth: how deep in the tree the node is
-           */
-          const horizontalCoefficient = Math.min(
-            item.value,
-            this.getLineLengthForDepth(item.depth),
-          );
-          const verticalCoefficient = Math.ceil(this.__getVerticalCoefficient(item));
-
-          item.x1 = item.x0 + (horizontalCoefficient * (this.minWidth + 2 * this.margin))
-            + (item.height * 2 * this.padding)
-            + (horizontalCoefficient - 1)
-            * (this.padding + 2 * this.margin);
-
-          item.y1 = item.y0
-            + (verticalCoefficient * this.minHeight)
-            + (item.height * this.padding)
-            + (verticalCoefficient - 1)
-            * (this.padding + this.margin);
-
-          if (item.data.drawOption) {
-            item.data.drawOption.needsResizing = false;
-            item.data.drawOption.width = item.x1 - item.x0;
-            item.data.drawOption.height = item.y1 - item.y0;
-          }
-        } else {
-          item.x1 = item.x0 + item.data.drawOption.width;
-          item.y1 = item.y0 + item.data.drawOption.height;
+    lines
+      .forEach((line) => {
+        let prevItem = {
+          x1: 0,
+          y0: line.band + this.padding,
+        };
+
+        line.items
+          .map((item) => {
+            if (!item.data.drawOption) {
+              item.data.drawOption = new ComponentDrawOption({
+                needsPositioning: true,
+                needsResizing: true,
+              });
+            }
+
+            return item;
+          })
+          .sort((itemA, itemB) => {
+            if (itemA.data.drawOption.needsPositioning && !itemB.data.drawOption.needsPositioning) {
+              return 1;
+            }
+
+            if (!itemA.data.drawOption.needsPositioning
+              && !itemB.data.drawOption.needsPositioning) {
+              return itemA.data.drawOption.x - itemB.data.drawOption.x;
+            }
+
+            return 0;
+          })
+          .forEach((item) => {
```

Comment:
isn't it possible to put the code of the `forEach` in an other function to reduce the size of `setupTiles` ?